tbtadm/controller.cpp

```diff
@@ -501,8 +501,10 @@ void tbtadm::Controller::approve(const fs::path& dir) try
         addToACL();
     }
 
+    bool hasKeyFile = fs::exists(keyFilename);
+
     std::ostringstream keyStream;
-    if (m_sl == 2)
+    if (m_sl == 2 && hasKeyFile)
```

Comment:
Following my comment from below, I think it's enough to add here `&& !m_once` to be able to connect legacy devices without adding them to ACL.

Reply:
Indeed, that seems to work. I'll update this patch accordingly, thanks!